scp.py

```diff
@@ -224,35 +245,36 @@ def _read_stats(self, name):
 
     def _send_files(self, files):
         for name in files:
-            basename = asbytes(os.path.basename(name))
             (mode, size, mtime, atime) = self._read_stats(name)
             if self.preserve_times:
                 self._send_time(mtime, atime)
-            file_hdl = open(name, 'rb')
-
-            # The protocol can't handle \n in the filename.
-            # Quote them as the control sequence \^J for now,
-            # which is how openssh handles it.
-            self.channel.sendall(("C%s %d " % (mode, size)).encode('ascii') +
-                                 basename.replace(b'\n', b'\\^J') + b"\n")
-            self._recv_confirm()
-            file_pos = 0
+            self._send_file(open(name, 'rb'), name, mode, size)
```

Comment:
You should probably still close the file handle explicitly after `_send_file()`

Reply:
@remram44 look at line 276, there's a `fl.close()` in `_send_file`, since this method always requires an open file handler in its input

Reply:
Having the function you give the file to close it is very unusual. It is probable that users of this will expect the file to remain open, or will close it a second time. Remember that a lot of files can be rewinded.

Reply:
ok, then I should close the file handle at line 252 and remove lin 276, correct?

Reply:
That is my opinion! Thanks